test/test_plugin_helper.rb

```diff
@@ -5,14 +5,14 @@
 FactoryBot.definition_file_paths << File.join(File.dirname(__FILE__), 'factories')
 # Add factories from foreman_ansible
 FactoryBot.definition_file_paths << File.join(ForemanAnsible::Engine.root, '/test/factories')
-FactoryBot.definition_file_paths << File.join(ForemanPuppet::Engine.root, '/test/factories') if defined?(ForemanPuppet)
+FactoryBot.definition_file_paths << File.join(ForemanPuppet::Engine.root, '/test/factories') if Foreman::Plugin.installed?("foreman_puppet")
```

Comment:
this is quite soon in the init proces and just for tests, I'd go with `defined?(ForemanPuppet::Engine`)` here, but probably both will work, so I'm fine both ways ^_^